app/docker/components/host-view-panels/node-labels-table/node-labels-table-controller.js

```diff
@@ -0,0 +1,39 @@
+angular.module('portainer.docker').controller('NodeLabelsTableController', [
+  function NodeLabelsTableController() {
+    var ctrl = this;
+    ctrl.removeLabel = removeLabel;
+    ctrl.updateLabel = updateLabel;
+
+    function removeLabel(index) {
+      var label = ctrl.labels.splice(index, 1);
+      if (label !== null) {
+      ctrl.onChangedLabels({ labels: ctrl.labels });
+      }
+    }
+
+    function updateLabel(label) {
+      if (
+        label.value !== label.originalValue ||
+        label.key !== label.originalKey
+      ) {
+        ctrl.onChangedLabels({ labels: ctrl.labels });
+      }
+    }
+
+   
+
+    function cancelChanges() {
```

Comment:
Seems that this function is not used. Should it be? Otherwise you can remove it.